langchain4j-agentic/src/main/java/dev/langchain4j/agentic/Agent.java

```diff
@@ -42,4 +42,11 @@
      * @return name of the output variable.
      */
     String outputName() default "";
+
+    /**
+     * If true, the agent will be invoked in a non-blocking manner, allowing the workflow to continue without waiting for the agent's result.
+     *
+     * @return true if the agent should be invoked in a non-blocking manner, false otherwise.
+     */
+    boolean nonBlocking() default false;
```

Comment:
```suggestion
    boolean blocking() default true;
```

Just a slight simplification, WDYT? If it makes sense, please rename it everywhere

Reply:
Maybe I'm overthinking this, but in my opinion it really depends on the default behavior of this feature. To be honest if I had thought to this possibility when I implemented the first release of the agentic module, I would have made all agents non-blocking by default. I'm not doing this now to avoid changing their behavior and breaking backward compatibility in a quite implicit and subtle way, even though I'm still tempted to change this.

In essence I see 2 alternatives:

1. I leave it as it is
2. I rename `nonBlocking` with `blocking` (and change the code accordingly), but also make the non-blocking behavior as the default one.

WDYT?

Reply:
I was referring only to the name of the `boolean` flag, not the default behaviour. AFAIK it is a good practice to avoid unnecessary negations when naming things to avoid additional mental overhead, but it is not really critical.

As for the default behaviour: if you are sure that non-blocking is the better default, I would say that it is still a good time to change it, as module is quite young and still experimental